envoy/config/filter/http/ext_authz/v2alpha/ext_authz.proto

```diff
@@ -0,0 +1,41 @@
+syntax = "proto3";
+
+package envoy.config.filter.http.ext_authz.v2alpha;
+option go_package = "v2";
+
+import "envoy/api/v2/core/grpc_service.proto";
+import "envoy/api/v2/core/http_uri.proto";
+
+import "google/protobuf/duration.proto";
+import "gogoproto/gogo.proto";
+
+// The external authorization HTTP service configuration.
+message HttpService {
+  // Sets the HTTP server URI which the authorization requests must be sent to.
+  envoy.api.v2.core.HttpUri server_uri = 1;
+
+  // Sets the time, in milliseconds, within the service should respond to an authorization
+  // request.
+  google.protobuf.Duration timeout = 2 [(gogoproto.stdduration) = true];
+
+  // Sets an optional prefix to the value of authorization request header `path`.
+  string path_prefix = 3;
+}
+
+message ExtAuthz {
+
+  oneof services {
+    // The external authorization gRPC service configuration.
+    envoy.api.v2.core.GrpcService grpc_service = 1;
+
+    // The external authorization HTTP service configuration.
+    HttpService http_service = 3;
```

Comment:
nit: should it be 2.

Reply:
Is it ok to re-index them? `bool failure_mode_allow = 3`?

Reply:
you are right `oneof` doesn't encapsulate the messages under it. We can leave it as is.